lighthouse/data-fixes/DPL-048/data_fix_and_save.py

```diff
@@ -1,15 +1,32 @@
-# get the root_sample_ids, fix them, write these to a CSV to be used with the 'write_data' script (which inserts the fixed IDs into the DBs)
+# get the data, fix it, write these to a CSV to be used with the 'write_data' script (which inserts the fixed data into the DBs)
+import pandas as pd
+import argparse
 
 from data_getters import get_data
 from data_helpers import remove_everything_after_first_underscore
 
-def save_data():
-    data = get_data()
-    print("Editing the root_sample_ids...")
-    data = data.rename(columns={"root_sample_id": "original_root_sample_id"})
-    data["root_sample_id"] = data["original_root_sample_id"].apply(remove_everything_after_first_underscore)
-    print("Adding the root_sample_ids to a CSV file.")
-    data.to_csv('data-fixes/test-data/root_sample_ids.csv', index=False)
+from constants import (
+    COLUMN_NAME,
+    ORIGINAL_COLUMN_NAME
+)
+
+def save_data(input_filename, output_filename):
+    if input_filename:
+        data = pd.read_csv(input_filename)
+    else:
+        data = get_data()
+
+    print("Editing the data...")
+    data = data.rename(columns={COLUMN_NAME: ORIGINAL_COLUMN_NAME})
+    data[COLUMN_NAME] = data[ORIGINAL_COLUMN_NAME].apply(remove_everything_after_first_underscore)
+    print("Adding the data to a CSV file.")
+    data.to_csv(output_filename, index=False)
 
 if __name__ == "__main__":
-    save_data()
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--input_file", required=False)
```

Comment:
`required=False`?  I think you later rely on there being a value.  Might be wrong.

Reply:
If you don't give an `input_file` it will return 'None' when you ask for it on line 30. Then that gets passed into the method, which has a check for whether it exists (ie. whether it's not None or it is None), and if it wasn't None it will read the file but if not it goes to the DB to get the data. This line was only for testing really, so that I could check that it would fix the data correctly and save to CSV when given some dummy data. In reality it should go to the DB to get the data because that's the data we're trying to fix

Reply:
Ahhh I see.  That makes sense!  Good stuff.